core/src/main/scala/org/apache/spark/util/VersionUtils.scala

```diff
@@ -63,4 +64,36 @@ private[spark] object VersionUtils {
           s" version string, but it could not find the major and minor version numbers.")
     }
   }
+
+  /**
+   * Extracts the major, minor and patch parts from the input `version`. Note that if minor or patch
+   * version is missing from the input, this will return 0 for these parts. Returns `None` if the
+   * input is not of a valid format.
+   *
+   * Examples of valid version:
+   *  - 1   (extracts to (1, 0, 0))
+   *  - 2.4   (extracts to (2, 4, 0))
+   *  - 3.2.2   (extracts to (3, 2, 2))
+   *  - 3.2.2.4   (extracts to 3, 2, 2))
+   *  - 3.3.1-SNAPSHOT   (extracts to (3, 2, 2))
```

Comment:
```scala
-   *  - 3.3.1-SNAPSHOT   (extracts to (3, 2, 2))
+   *  - 3.2.1-SNAPSHOT   (extracts to (3, 2, 2))
```

Reply:
```
*  - 3.2.1-SNAPSHOT   (extracts to (3, 2, 1))?
```

Reply:
oops - nice catch!

Reply:
I updated it to `*  - 3.3.1-SNAPSHOT   (extracts to (3, 3, 1))`